hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java

```diff
@@ -165,16 +165,20 @@
         Map<String, List<HoodieBaseFile>> groupedInputSplits = partitionsToParquetSplits.get(partitionPath).stream()
             .collect(Collectors.groupingBy(file -> FSUtils.getFileId(file.getFileStatus().getPath().getName())));
         latestFileSlices.forEach(fileSlice -> {
-          List<HoodieBaseFile> dataFileSplits = groupedInputSplits.get(fileSlice.getFileId());
-          dataFileSplits.forEach(split -> {
-            try {
-              List<String> logFilePaths = fileSlice.getLogFiles().sorted(HoodieLogFile.getLogFileComparator())
-                  .map(logFile -> logFile.getPath().toString()).collect(Collectors.toList());
-              resultMap.put(split, logFilePaths);
-            } catch (Exception e) {
-              throw new HoodieException("Error creating hoodie real time split ", e);
-            }
-          });
+          final String fileId = fileSlice.getFileId();
+          // filter out the file group that has only logs (say the index is global).
+          if (groupedInputSplits.containsKey(fileId)) {
+            List<HoodieBaseFile> dataFileSplits = groupedInputSplits.get(fileId);
+            dataFileSplits.forEach(split -> {
+              try {
+                List<String> logFilePaths = fileSlice.getLogFiles().sorted(HoodieLogFile.getLogFileComparator())
+                    .map(logFile -> logFile.getPath().toString()).collect(Collectors.toList());
+                resultMap.put(split, logFilePaths);
+              } catch (Exception e) {
+                throw new HoodieException("Error creating hoodie real time split ", e);
```

Comment:
nit : Should we modify the exception message so that when throw an exception, we can distinguish it from the exception in the `getRealtimeSplits` method?

Reply:
The exception already throws out, what kind of exception do you suggest it to be ?

Reply:
I mean to modify this content，for example 'Error creating hoodie real time split for group logs by BaseFile',because the exception content of `groupLogsByBaseFile` and `getRealtimeSplits` method are the same.

Reply:
> When we use hive query the hudi table, the same NPE exception may also occur [here](https://github.com/apache/hudi/blob/master/hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/utils/HoodieRealtimeInputFormatUtils.java#L105) , should we add the same judgment logic?

Oops, i guess we should